src/mrack/providers/openstack.py

```diff
@@ -117,8 +121,102 @@ async def _openstack_gather_responses(self, *calls):
 
         return result
 
+    async def _create_session(self):
+        """
+        Create session object using credentials.
+
+        Credentials are retrieved from either enviromental variables or
+        clouds.yaml file.
+        Both username+password and application credentials formats
+        are accepted.
+
+        Returns: Session object.
+        """
+        try:
+            # Create session from environment variables
+            return AuthPassword()
+        except TypeError:
+            # Fallback to create session from clouds.yaml file
+            try:
+                config = os_client_config.OpenStackConfig()
+
+                # Get the cloud specified in provisioning-config
+                # If that fails try to pick the cloud specified in OS_CLOUD envvar
+                # or get the only cloud (if there is only one)
+                try:
+                    cloud = config.get_one_cloud(self.cloud_profile)
+                except os_client_config.exceptions.OpenStackConfigException:
+                    cloud = config.get_one_cloud()
+
+                auth_info = cloud.config["auth"]
```

Comment:
I'd add bit more logging here. E.g. when I was testing it, I did not know from the logs what credentials it actually picked so I had to use:

```diff
diff --git a/src/mrack/providers/openstack.py b/src/mrack/providers/openstack.py
index cd9613c..0d81d6e 100644
--- a/src/mrack/providers/openstack.py
+++ b/src/mrack/providers/openstack.py
@@ -149,8 +149,10 @@ class OpenStackProvider(Provider):
                     cloud = config.get_one_cloud()
 
                 auth_info = cloud.config["auth"]
+                logger.debug(f"OpenStack auth_url: {auth_info['auth_url']}")
 
                 if "username" in auth_info and "password" in auth_info:
+                    logger.debug(f"OpenStack username: {auth_info['username']}"),
                     return AuthPassword(
                         auth_url=auth_info["auth_url"],
                         username=auth_info["username"],
@@ -162,6 +164,7 @@ class OpenStackProvider(Provider):
                     "application_credential_id" in auth_info
                     and "application_credential_secret" in auth_info
                 ):
+                    logger.debug(f"OpenStack application_credential_id: {auth_info['application_credential_id']}")
                     return AuthPassword(
                         auth_url=auth_info["auth_url"],
                         application_credential_id=auth_info[
@@ -183,6 +186,7 @@ class OpenStackProvider(Provider):
                 OptionError,
                 MissingRequiredOptions,
             ) as terr:
+                logger.debug(f"Error during authentication: {terr}")
                 err_msg = (
                     "OpenStack credentials not provided or not properly configured. "
                     + "Authentication failed."
```

Such logging could be useful also when debugging issues in production.

Reply:
Logging messages added